### PR TITLE
[+] Acts on taggable - Search & Filters

### PR DESCRIPTION
**Pull request by SeyZ**
Not sure about the implementation, would love to have your feedback @arnaudbesnier and @lucasscariot

### CODE REVIEW

**Comment by SeyZ** [comment]
If you agree with the implementation, do you think it's easy to only allow "contains" filter on the UI @arnaudbesnier ?

**Comment by arnaudbesnier** [comment]
The `tags` field seems to be save with a `String` type in Forest database (according to @lucasscariot) so I don't think it will be easy to do that.

File: app/services/forest_liana/search_query_builder.rb [hunk 1]
```diff
@@ -79,13 +79,29 @@ def search_param
...
+
         @params[:filter].each do |field, values|
-          next if association?(field)
+          # ActsAsTaggable
+          if acts_as_taggable?(field)
+            tagged_records = @records.tagged_with(values.tr('*', ''))
+
+            if @params[:filterType] == 'and'
+              @records = tagged_records
+            elsif @params[:filterType] == 'or'
```
Comment (arnaudbesnier) [inline app/services/forest_liana/search_query_builder.rb]: If you can do that for the filters, why not doing it this way (double request) for the search?

File: app/services/forest_liana/search_query_builder.rb [hunk 2]
```diff
@@ -79,13 +79,29 @@ def search_param
...
+              ids = tagged_records
+                .map {|t| t[@resource.primary_key]}
+                .join(',')
+
+              conditions << "#{@resource.primary_key} IN (#{ids})"
```
Comment (arnaudbesnier) [inline app/services/forest_liana/search_query_builder.rb]: Shouldn't you escape the column name to prevent issues?

**Comment by SeyZ** [comment]
[(2) Users want to search on a "tags" field created using the "acts_as_taggable" gem](https://trello.com/c/Vojc8RtZ/1421-2-users-want-to-search-on-a-tags-field-created-using-the-acts-as-taggable-gem)

File: app/services/forest_liana/search_query_builder.rb [hunk 3]
```diff
@@ -34,6 +34,16 @@ def format_column_name(table_name, column_name)
       ForestLiana::AdapterHelper.format_column_name(table_name, column_name)
     end
 
+    def acts_as_taggable_query(tagged_records)
+      ids = tagged_records
+        .map {|t| t[@resource.primary_key]}
+        .join(',')
+
+      if ids.present?
```
Comment (arnaudbesnier) [inline app/services/forest_liana/search_query_builder.rb]: On a search, If there is no `ids` found (no record tagged with the search value), the liana crashes.

**Pull request merged**